tests/components/image_processing/test_init.py

```diff
@@ -7,286 +7,256 @@
 
-from tests.common import (
-    assert_setup_component,
-    get_test_home_assistant,
-    get_test_instance_port,
-)
+from tests.common import assert_setup_component, get_test_instance_port
 from tests.components.image_processing import common
 
 
-class TestSetupImageProcessing:
-    """Test class for setup image processing."""
+def get_url(hass):
+    """Return camera url."""
+    state = hass.states.get("camera.demo_camera")
+    return f"{hass.config.internal_url}{state.attributes.get(ATTR_ENTITY_PICTURE)}"
 
-    def setup_method(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
 
-    def teardown_method(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+async def setup_image_processing(hass):
+    """Set up things to be run when tests are started."""
+    hass.data.pop(DATA_CUSTOM_COMPONENTS)
 
-    def test_setup_component(self):
-        """Set up demo platform on image_process component."""
-        config = {ip.DOMAIN: {"platform": "demo"}}
+    await async_setup_component(
+        hass,
+        http.DOMAIN,
+        {http.DOMAIN: {http.CONF_SERVER_PORT: get_test_instance_port()}},
+    )
 
-        with assert_setup_component(1, ip.DOMAIN):
-            setup_component(self.hass, ip.DOMAIN, config)
+    config = {ip.DOMAIN: {"platform": "test"}, "camera": {"platform": "demo"}}
 
-    def test_setup_component_with_service(self):
-        """Set up demo platform on image_process component test service."""
-        config = {ip.DOMAIN: {"platform": "demo"}}
+    await async_setup_component(hass, ip.DOMAIN, config)
+    await hass.async_block_till_done()
 
-        with assert_setup_component(1, ip.DOMAIN):
-            setup_component(self.hass, ip.DOMAIN, config)
 
-        assert self.hass.services.has_service(ip.DOMAIN, "scan")
+async def setup_image_processing_alpr(hass):
+    """Set up things to be run when tests are started."""
+    config = {ip.DOMAIN: {"platform": "demo"}, "camera": {"platform": "demo"}}
 
+    with patch(
+        "homeassistant.components.demo.image_processing."
+        "DemoImageProcessingAlpr.should_poll",
+        new_callable=PropertyMock(return_value=False),
+    ):
+        await async_setup_component(hass, ip.DOMAIN, config)
+        await hass.async_block_till_done()
 
-class TestImageProcessing:
-    """Test class for image processing."""
+    alpr_events = []
 
-    def setup_method(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.hass.data.pop(DATA_CUSTOM_COMPONENTS)
+    @callback
+    def mock_alpr_event(event):
+        """Mock event."""
+        alpr_events.append(event)
 
-        setup_component(
-            self.hass,
-            http.DOMAIN,
-            {http.DOMAIN: {http.CONF_SERVER_PORT: get_test_instance_port()}},
-        )
+    hass.bus.async_listen("image_processing.found_plate", mock_alpr_event)
```

Comment:
We can use this:
https://github.com/home-assistant/core/blob/5c717cbb1d9a0f9c19da06173520c51cf6ddec47/tests/common.py#L1119-L1129

Reply:
Fixed in 0816a0b